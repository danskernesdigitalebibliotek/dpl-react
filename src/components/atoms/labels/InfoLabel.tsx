import * as React from "react";
import { FC } from "react";

export interface InfoLabelProps {
  children: React.ReactNode;
}

const InfoLabel: FC<InfoLabelProps> = ({ children }) => {
  return <div className="status-label status-label--info">{children}</div>;
};

export default InfoLabel;
