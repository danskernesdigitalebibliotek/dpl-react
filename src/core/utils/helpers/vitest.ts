export const isVitestEnvironment =
  typeof process !== "undefined" && process.env && process.env.VITEST;

export default {};
