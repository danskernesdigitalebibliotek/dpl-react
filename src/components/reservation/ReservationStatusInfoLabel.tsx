import * as React from "react";
import { FC } from "react";
import InfoLabel from "../atoms/labels/InfoLabel";
import { useText } from "../../core/utils/text";
import { getReservationStatusInfoLabel } from "../../apps/reservation-list/utils/helpers";

export interface ReservationStatusInfoLabelProps {
  pickupBranch?: string | null;
  date: string;
  isDigital: boolean;
}

const ReservationStatusInfoLabel: FC<ReservationStatusInfoLabelProps> = ({
  pickupBranch,
  date,
  isDigital
}) => {
  const t = useText();
  return (
    <InfoLabel dataCy="reservation-status-info-label">
      {getReservationStatusInfoLabel({
        pickupBranch,
        date,
        t,
        isDigital
      })}
    </InfoLabel>
  );
};

export default ReservationStatusInfoLabel;
