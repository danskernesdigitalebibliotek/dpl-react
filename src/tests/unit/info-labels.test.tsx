import { describe, expect, it } from "vitest";
import React, { ReactNode } from "react";
import { configure, render } from "@testing-library/react";
import { combineReducers, configureStore } from "@reduxjs/toolkit";
import { Provider } from "react-redux";
import InfoLabel from "../../components/atoms/labels/InfoLabel";
import ReservationStatusInfoLabel from "../../components/reservation/ReservationStatusInfoLabel";
import textReducer from "../../core/text.slice";

configure({
  testIdAttribute: "data-cy"
});

const store = configureStore({
  reducer: combineReducers({
    text: textReducer
  }),
  preloadedState: {
    text: {
      data: {
        reservationListLoanBeforeText: "Borrow before @date"
      }
    }
  }
});

const Wrapper = ({ children }: { children: ReactNode }) => (
  <Provider store={store}> {children} </Provider>
);

describe("InfoLabel", () => {
  it("Should show the markup for the simple info label", async () => {
    const { getByTestId } = render(<InfoLabel>Some text</InfoLabel>);

    const label = getByTestId("info-label");

    expect(label).toMatchSnapshot();
  });
});

describe("ReservationStatusInfoLabel", () => {
  it("Should show the markup for the for a reservation status for a reservation status info label", async () => {
    const { getByText } = render(
      <Wrapper>
        <ReservationStatusInfoLabel date="12-12-22 12:12" isDigital={false} />
      </Wrapper>
    );

    const label = getByText("Borrow before 12-12-2022");

    expect(label).toMatchSnapshot();
  });

  it("Should show the markup for the for a reservation status for a DIGITAL reservation status info label", async () => {
    const { getByText } = render(
      <Wrapper>
        <ReservationStatusInfoLabel date="12-12-22 12:12" isDigital />
      </Wrapper>
    );

    const label = getByText("Borrow before 12-12-2022 12:12");

    expect(label).toMatchSnapshot();
  });
});
