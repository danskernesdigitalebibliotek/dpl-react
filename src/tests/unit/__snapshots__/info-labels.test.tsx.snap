// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`InfoLabel > Should show the markup for the simple info label 1`] = `
<div
  class="status-label status-label--info"
  data-cy="info-label"
>
  Some text
</div>
`;

exports[`ReservationStatusInfoLabel > Should show the markup for the for a reservation status for a DIGITAL reservation status info label 1`] = `
<div
  class="status-label status-label--info"
  data-cy="reservation-status-info-label"
>
  Borrow before 12-12-2022 12:12
</div>
`;

exports[`ReservationStatusInfoLabel > Should show the markup for the for a reservation status for a reservation status info label 1`] = `
<div
  class="status-label status-label--info"
  data-cy="reservation-status-info-label"
>
  Borrow before 12-12-2022
</div>
`;
